src/util/mod-embed-factory.ts

```diff
@@ -0,0 +1,98 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import { EmbedBuilder, Guild, GuildMember, User } from "discord.js";
+import { GREEN, RED, YELLOW } from "./colours";
+
+export enum Action {
+    KICK,
+    BAN,
+    UNBAN
+}
+
+/**
+ * An embed factory for kicking, banning, and unbanning users
+ * 
+ * @param user the Discord user this action concerns
+ * @param server the server this action is performed in
+ * @param action the type of action performed
+ * @param reason the reason for the action
+ */
+export const genEmbed = (user: User | GuildMember, server: Guild, action: Action, reason: string = null): EmbedBuilder => {
+    // The embed for the moderation action
+    const embed: EmbedBuilder = new EmbedBuilder();
+
+    // Constant for converting millisecond timestamps to regular timestamps
+    const millisToSecs: number = 1000;
+
+    // Set the title and colour of the embed depending on the action
+    switch (action) {
+        case Action.KICK:
+            embed.setAuthor({ name: "User kicked", iconURL: server.iconURL() })
+                .setColor(YELLOW);
+            break;
+        case Action.BAN:
+            embed.setAuthor({ name: "User banned", iconURL: server.iconURL() })
+                .setColor(RED);
+            break;
+        case Action.UNBAN:
+            embed.setAuthor({ name: "User unbanned", iconURL: server.iconURL() })
+                .setColor(GREEN);
+            break;
+    }
+
+    // Convert member objects to user objects, to access the username
+    const userObj: User = user instanceof GuildMember ? user.user : user;
+
+    // Username displayed in the embed
+    const username: string = userObj.username;
+    // The user discriminator
+    const discriminator: string = userObj.discriminator;
+
+    // Fill in the rest on the embed
+    embed.setTitle(`${username}${discriminator === "0" ? "" : `#${discriminator}`}`)
+        .setThumbnail(userObj.avatarURL())
+        .addFields(
+            {
+                name: "User ID", value: `${userObj.id}`, inline: true
+            },
+            {
+                name: "Created", value: `<t:${Math.floor(userObj.createdTimestamp / millisToSecs)}>`, inline: true
+            },
+            {
+                name: "Joined", value: user instanceof GuildMember ? `<t:${Math.floor(user.joinedTimestamp / millisToSecs)}>` : "N/A", inline: true
+            },
+            {
+                name: "Reason", value: reason ?? "N/A", inline: true
+            },
+            {
+                name: "Bot user", value: `${userObj.bot}`, inline: true
+            },
+            {
+                name: "System user", value: `${userObj.system}`, inline: true
+            }
+        );
+
+    return embed;
+};
```

Comment:
This module exists so the same lines of code doesn't have to be repeated in the 3 moderation commands